src-tauri/src/main.rs

```diff
@@ -1,8 +1,35 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+use std::path::Path;
+
+use tauri::api::path::home_dir;
+
 fn main() {
   tauri::Builder::default()
+    .invoke_handler(tauri::generate_handler![
+      list
+      ])
     .run(tauri::generate_context!())
     .expect("error while running tauri application");
 }
+
+/// list all possible README.md locations in dirs of the save dir
+#[tauri::command]
+async fn list() -> Result<Vec<String>, String> {
+  let base_path = Path::new(home_dir().unwrap().as_path()).join(".notes");
+  
+  let mut notes = Vec::<String>::new();
+
+  // read all files in dir
+  let dir_entries = std::fs::read_dir(base_path).map_err(|err| err.to_string())?;
+  for dir_entry in dir_entries {
```

Comment:
It would be great to have a list of found dirs and return a list of errors for all error while reading too so the frontend can decide, whether it wants to show the errors for each "not processible" file or just throw this away

Reply:
again, tests would be great for this

Reply:
> [tauri documentation](https://tauri.app/v1/guides/features/command/) recommends to use Result enums to return errors.

I would therefore never return a Result enum as this would be boiler

Reply:
> For me this sounds like you want to list all correctly structured directories and if on any of the operations an error occurs, the result doesn't list the directory. Up to the point where the result is empty because the base dir is not present. The Result would never have the error string filled in. Every invoke would be without error.

Yes indeed. When the base directory is not present, we should create it. I think, we could do this on startup. Note that it is normal that the base directory does not exist at the very first start.

Reply:
> > [tauri documentation](https://tauri.app/v1/guides/features/command/) recommends to use Result enums to return errors.
> 
> I would therefore never return a Result enum as this would be boiler

This is fine by me.